src/containers/Tag/TagList/TagList.tsx

```diff
@@ -53,67 +121,154 @@ export const TagList: React.SFC<TagListProps> = (props) => {
+    // Should be type tag, but can't import Tag type into file
+    return tags.map((t: any) => {
+      return {
+        label: t.label,
+        description: t.description,
+        operations: getIcons(t.id),
+      };
     });
-  } else {
-    listing = (
-      <TableRow>
-        <TableCell>There are no tags.</TableCell>
-      </TableRow>
-    );
+  }
+
+  const resetTableVals = () => {
+    setTableVals({
+      pageNum: 0,
+      pageRows: 10,
+      sortCol: columnNames[0],
+      sortDirection: 'asc',
+    });
+  };
+
+  const handleSearch = (e: any) => {
+    e.preventDefault();
+    let searchVal = e.target.nameSearch.value.trim();
+    setSearchVal(searchVal);
+    resetTableVals();
+  };
+
+  // Making different columns for the table
+  const oldColumns = [
+    {
+      name: 'Name',
+      selector: 'label',
+      sortable: true,
+    },
+    {
+      name: 'Description',
+      selector: 'description',
+      sortable: false,
+    },
+    {
+      name: 'Operations',
+      selector: 'operations',
+      sortable: false,
+    },
+  ];
+
+  // Get tag data and total number of tags.
+  let tagList: any;
+  if (data) {
+    tagList = formatTags(data.tags);
+  }
+
+  let tagCount: number = tableVals.pageRows;
+  if (countData) {
+    tagCount = countData.countTags;
   }
 
   return (
-    <div>
-      <div className={styles.AddButtton}>
-        <Button variant="contained" color="primary" onClick={() => setNewTag(true)}>
-          New Tag
-        </Button>
+    <>
+      <div className={styles.Header}>
+        <Typography variant="h5" className={styles.Title}>
+          Tags
+        </Typography>
+        <div className={styles.Buttons}>
+          <IconButton className={styles.IconButton} onClick={() => setSearchOpen(!searchOpen)}>
+            <SearchIcon className={styles.SearchIcon}></SearchIcon>
+          </IconButton>
+          <form onSubmit={handleSearch}>
+            <div className={searchOpen ? styles.SearchBar : styles.HideSearchBar}>
+              <InputBase
+                defaultValue={searchVal}
+                className={searchOpen ? styles.ShowSearch : styles.HideSearch}
+                name="nameSearch"
+              />
+              {searchOpen ? (
+                <div
+                  className={styles.ResetSearch}
+                  onClick={() => {
+                    setSearchVal('');
+                    resetTableVals();
+                  }}
+                >
+                  <Divider orientation="vertical" />
+                  <CloseIcon className={styles.CloseIcon}></CloseIcon>
+                </div>
+              ) : null}
+            </div>
+          </form>
+          <Button
+            color="primary"
+            variant="contained"
+            disableElevation={true}
+            onClick={() => setNewTag(true)}
+          >
+            Add New
+          </Button>
+        </div>
       </div>
+
+      {/* Rendering list of tags */}
```

Comment:
This is table with client-side rendering (which will be removed later, just here for testing purposes).